experiments/AMIP/modular/coupler_driver_modular.jl

```diff
@@ -407,14 +426,26 @@ step!(land_sim, Δt_cpl)
 step!(ocean_sim, Δt_cpl)
 step!(ice_sim, Δt_cpl)
 
-# 3) coupler re-imports updated surface fields and calculates turbulent fluxes on the atmos grid, while updating atmos sfc_conditions
-import_combined_surface_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes) # i.e. T_sfc, albedo, z0, beta
-compute_combined_turbulent_fluxes!(cs.model_sims, cs.fields, turbulent_fluxes)
+# 3) coupler re-imports updated surface fields and calculates turbulent fluxes, while updating atmos sfc_conditions
+if turbulent_fluxes isa CombinedAtmosGrid
+    # calculate combined fluxes on the atmos grid
+    import_combined_surface_fields!(cs.fields, cs.model_sims, cs.boundary_space, turbulent_fluxes) # i.e. T_sfc, albedo, z0, beta, q_sfc
+    combined_turbulent_fluxes!(cs.model_sims, cs.fields, turbulent_fluxes) # this updates the sfc_ts
+elseif turbulent_fluxes isa PartitionedComponentModelGrid
+    # calculate turbulent fluxes in surface models and save the weighted average in coupler fields
+    partitioned_turbulent_fluxes!(cs.model_sims, cs.fields, cs.boundary_space, MoninObukhovScheme(), thermo_params)
+
+    # update atmos sfc_conditions for surfce temperature
+    # TODO: this isd hard coded and needs to be simplified (need CA modification)
+    new_p = get_new_p(atmos_sim, cs.fields)
+    ClimaAtmos.SurfaceConditions.update_surface_conditions!(atmos_sim.integrator.u, new_p, atmos_sim.integrator.t) # sets T_sfc (but SF calculation not necessary - CA)
+    atmos_sim.integrator.p.sfc_conditions .= new_p.sfc_conditions
+end
 
 # 4) given the new sfc_conditions, atmos calls the radiative flux callback
 reinit!(atmos_sim) # sets a nonzero radiation flux
 
-# 5) coupler re-imports updated atmos fluxes, and sends them to surface models
+# 5) coupler re-imports updated atmos fluxes, and sends them to surface models (radiative to surfaces, and turbulent to all)
```

Comment:
This says the fluxes are sent to surface models, then says radiative are sent to surfaces and turbulent to all. Should it instead say that the fluxes are send to component models (and then further specify from there)?

Reply:
"surface models" is more accurate. I can clarify this. But yes, something we need to work on with Atmos to simplify the init stage.